sentinel-ai-models-simple-openai/src/main/java/com/phonepe/sentinelai/models/SimpleOpenAIModel.java

```diff
@@ -1118,4 +1123,43 @@ private ToolChoiceOption computeToolChoice(OutputGenerationMode outputGeneration
         };
     }
 
+    private void executeMessagesPreProcessors(final List<AgentMessagesPreProcessor> messagesPreProcessors,
+                                              final ModelUsageStats statsForRun,
+                                              final ModelRunContext context,
+                                              final ArrayList<AgentMessage> allMessages,
```

Comment:
Don't pass implementations as parameters unless you are planning to use some special behaviour in the implementation